plugins/poll/assets/javascripts/discourse/components/poll-option-irv-dropdown.gjs

```diff
@@ -0,0 +1,58 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { fn } from "@ember/helper";
+import { action } from "@ember/object";
+import DButton from "discourse/components/d-button";
+import DropdownMenu from "discourse/components/dropdown-menu";
+import icon from "discourse-common/helpers/d-icon";
+import I18n from "discourse-i18n";
+import DMenu from "float-kit/components/d-menu";
+
+export default class PollOptionsDropdownComponent extends Component {
+  @tracked rank;
+
+  constructor() {
+    super(...arguments);
+    this.rank = this.args.rank;
+  }
+
+  @action
+  onRegisterApi(api) {
+    this.dMenu = api;
+  }
+
+  @action
+  selectRank(option, rank) {
+    this.args.sendRank(option, rank);
+    this.rank = rank === 0 ? I18n.t("poll.options.irv.abstain") : rank;
```

Comment:
We don't want to mutate an external variable from inside the component, this is a violation of DDAU (data down actions up)

Reply:
@jjaffeux but this is currently a local variable, so this is legal?